pkg/backend/display/ai.go

```diff
@@ -0,0 +1,238 @@
+	"github.com/pulumi/pulumi/sdk/v3/go/common/env"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+var atlasBaseURLs = map[string]string{
+	"https://api.pulumi.com": "https://app.pulumi.com",
+
+	// FIXME(dev): Remove before merging
+	// Local instance logged into using `pulumi login https://api.simon-local.pulumi.local`
+	"https://api.simon-local.pulumi.local": "https://app.simon-local.pulumi.local",
+}
+
+const (
+	atlasApiPath = "/pulumi-ai/atlas/api/ai/chat/preview"
+
+	// Environment variable to override the Atlas base URL for local development
+	debugAtlasBaseVar = "DEBUG_PULUMI_ATLAS_BASE"
+)
+
+// Matching the TypeScript interface structure
+type SummarizeUpdate struct {
+	Summary string `json:"summary"`
+}
+
+type Message struct {
+	Role    string          `json:"role"`
+	Kind    string          `json:"kind"`
+	Content json.RawMessage `json:"content"`
+}
+
+type AtlasUpdateSummaryResponse struct {
+	Messages []Message `json:"messages"`
+	Error    string    `json:"error"`
+	Details  any       `json:"details"`
+}
+
+type CloudContext struct {
+	OrgId string `json:"orgId"`
+	Url   string `json:"url"`
+}
+
+type ClientState struct {
+	CloudContext CloudContext `json:"cloudContext"`
+}
+
+type State struct {
+	Client ClientState `json:"client"`
+}
+
+type SkillParams struct {
+	PulumiUpdateOutput string `json:"pulumiUpdateOutput"`
+}
+
+type DirectSkillCall struct {
+	Skill  string      `json:"skill"`
+	Params SkillParams `json:"params"`
+}
+
+type AtlasUpdateSummaryRequest struct {
+	Query           string          `json:"query"`
+	State           State           `json:"state"`
+	DirectSkillCall DirectSkillCall `json:"directSkillCall"`
+}
+
+func getCurrentCloudURL() (string, error) {
+	ws := pkgWorkspace.Instance
+
+	project, _, err := ws.ReadProject()
+	if err != nil {
+		return "", fmt.Errorf("looking up project: %w", err)
+	}
+
+	url, err := pkgWorkspace.GetCurrentCloudURL(ws, env.Global(), project)
+	if err != nil {
+		return "", fmt.Errorf("could not get cloud url: %w", err)
+	}
+
+	return url, nil
+}
+
+// getSummaryToken retrieves the authentication token for the Atlas API
+func getSummaryToken(cloudUrl string) (string, error) {
+	account, err := workspace.GetAccount(cloudUrl)
+	if err != nil {
+		return "", fmt.Errorf("getting account: %w", err)
+	}
+
+	if account.AccessToken == "" {
+		return "", fmt.Errorf("no access token found for %s", cloudUrl)
+	}
+
+	return account.AccessToken, nil
+}
+
+// TODO(atlas): Replace with actual org ID from stack context
+// Using pulumi_local as temporary default for development
+func getOrgID() (string, error) {
+	return "pulumi_local", nil
+}
+
+// createAtlasRequest creates a new AtlasUpdateSummaryRequest with the given content and org ID
+func createAtlasRequest(content string, orgID string) AtlasUpdateSummaryRequest {
+	return AtlasUpdateSummaryRequest{
+		Query: "FIXME in atlas, this is ignored, but still required by zod",
```

Comment:
Good point. We still want to require it for non-direct calls, but the enforcement can be outside of Zod.